### ISSUE TITLE
CodeSnapASCII not calculating whitespace correctly

### ISSUE THREAD

**Issue by WizardStark** [mistricky/codesnap.nvim#119]
See below snippet from my config

```
╭────────────────────────────────────────────────────────────────────────────────────────╮
│ nvim/lua/workspaces/persistence.lua                                                    │
│────────────────────────────────────────────────────────────────────────────────────────│
│ 193 local function load_marks()                                                        │
│ 194 	local marks_file = Path:new(M.workspaces_path .. Path.path.sep .. "marks.json")   │
│ 195 	local marks_data = nil                                                            │
│ 196 	if marks_file:exists() then                                                       │
│ 197 		marks_data = vim.fn.json_decode(marks_file:read())                               │
│ 198 	end                                                                               │
│ 199                                                                                    │
│ 200 	if not marks_data then                                                            │
│ 201 		return                                                                           │
│ 202 	end                                                                               │
│ 203                                                                                    │
│ 204 	state.get().marks = marks_data                                                    │
│ 205 end                                                                                │
╰────────────────────────────────────────────────────────────────────────────────────────╯
```
Also breaks in go
```
╭────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Pack/git/git.go                                                                                        │
│────────────────────────────────────────────────────────────────────────────────────────────────────────│
│ 39 func AsyncGitRemoteRefExists(s *config.Session, packageDir string) chan GitRemoteRefExistsResult {  │
│ 40 	res := make(chan GitRemoteRefExistsResult)                                                         │
│ 41 	go func() {                                                                                        │
│ 42 		exists, err := GitRemoteRefExists(s, packageDir)                                                  │
│ 43 		res <- GitRemoteRefExistsResult{exists, err}                                                      │
│ 44 		close(res)                                                                                        │
│ 45 	}()                                                                                                │
│ 46                                                                                                     │
│ 47 	return res                                                                                         │
│ 48 }                                                                                                   │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────╯
```

In my `.zshrc` this works
```
╭─────────────────────────────────────────────────╮
│ dotfiles/home/.zshrc                            │
│─────────────────────────────────────────────────│
│ 123 [ -f ~/.lcl.zshrc ] && source ~/.lcl.zshrc  │
│ 124                                             │
│ 125 if [ -d ~/dotfile-shards/ ]; then           │
│ 126   for f in ~/dotfile-shards/*; do           │
│ 127     source $f                               │
│ 128   done                                      │
│ 129 fi                                          │
╰─────────────────────────────────────────────────╯
```

The only difference between these as far as I can tell is that the `.zshrc` is indented with tabs instead of spaces.

Loving the image snapshots, but this ASCII snapshot is the killer feature for me, so i'd love to get it working.

My setup function:
```
require("codesnap").setup({
	breadcrumbs_separator = "/",
	has_breadcrumbs = true,
	has_line_number = true,
	show_workspace = true,
	bg_padding = 0,
	watermark = "",
})
```

**Comment by mistricky** [comment]
Hi @WizardStark, I'm so glad to hear that you like the `ASCII snapshot` feature, it's really cool isn't it? XD 

CodeSnap indeed has issues handling whitespace in ASCII snapshot, I have some fixes, can you test with v1.5.2?

**Comment by WizardStark** [comment]
Perfect, thanks!
